app/models/order_item.rb

```diff
@@ -11,19 +11,16 @@ class OrderItem < ActiveRecord::Base
   validates :product_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :order_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :quantity_ordered, presence: true, numericality: { only_integer: true, greater_than: 0 }
-  validate :order_item_is_unique? # OPTIMIZE: I think this makes the above before_create obsolete?!
+  validate :order_item_is_unique?
 
 
+  # mutative methods
+
   def more!
-    reload # removing this line == DANGER WILL ROBINSON
-    # if the OrderItem isn't reloaded, ln30 will resolve based on a cached operation
-    # in other words, if line 30 has recently been evaluated for this OrderItem,
-    # then __it will not be evaluated__ and the cached value (true) will be used
-    # instead. by reloading, we force a new SQL query to be run to check anew
-    # whether product_has_stock?
-
-    if product_has_stock?
+    if product_has_stock? && product.stock > quantity_ordered
```

Comment:
this is not the same as `product.stock?`, which only checks whether a product's stock is greater than zero. this method is also checking whether a product's stock is greater than the quantity desired by the customer.